### PR TITLE
adding a notebook to show different ways to load text in tensorflow 

### PR DESCRIPTION
**Pull request by sanjanalreddy**
Adding a notebook for the NLP deep dive to show how to load text in tensorflow

In order to run the notebook:
- Use a Jupyter Lab notebook instance with TF2.8
- Install tensorflow text using `pip install "tensorflow-text==2.8.*"` (Please ensure you are installing tensorflow-text 2.8, if you install without specifying the version number it will corrupt you instance)

This lab is a modified version of this [tensorflow tutorial ](https://www.tensorflow.org/tutorials/load_data/text)

### CODE REVIEW

File: notebooks/text_models/solutions/load_text.ipynb [hunk 1]
```diff
@@ -0,0 +1,1935 @@
+{
```
Comment (takumiohym) [inline notebooks/text_models/solutions/load_text.ipynb:1]: <p>It would be helpful to mention that <code>tf.keras.utils.text_dataset_from_directory</code> uses directory names as label names automatically, and it return tuples of <code>(feature, label)</code>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/257/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (takumiohym) [inline notebooks/text_models/solutions/load_text.ipynb:1]: <p>Maybe explain why we set seed here?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/257/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (takumiohym) [inline notebooks/text_models/solutions/load_text.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.            print("Label:", label_batch.numpy()[i])
</pre><p>I would add s separater after this like <code>print("*" * 50)</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/257/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (takumiohym) [inline notebooks/text_models/solutions/load_text.ipynb:1]: <p>Very clear!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/257/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (BenoitDherin) [reply notebooks/text_models/solutions/load_text.ipynb:1]: <p>+1</p>

Comment (BenoitDherin) [reply notebooks/text_models/solutions/load_text.ipynb:1]: <p>If we can fix the seed at the beginning and not in the local function that would be better. Maybe by using something like the following at the beginning of the file?</p><p> </p><pre class="ql-syntax" spellcheck="false">SEED = 0
tf.keras.utils.set_random_seed(
    SEED
)
</pre>

Comment (BenoitDherin) [reply notebooks/text_models/solutions/load_text.ipynb:1]: <p>+1 (the output is a bit hard to read)</p>

[the remark this answers is not in the document]
Comment (BenoitDherin) [reply notebooks/text_models/solutions/load_text.ipynb:1]: <p>+1</p>

Comment (BenoitDherin) [inline notebooks/text_models/solutions/load_text.ipynb:1]: <p>What about a more descriptive title? For example:</p><p>"Loading and pre-processing text with Keras"</p><p>"Building input pipelines with Keras for NLP"</p><p>"Building NLP input pipelines with Keras and Tensorflow"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/257/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (BenoitDherin) [inline notebooks/text_models/solutions/load_text.ipynb:1]: <ul><li>Some missing "Learn the difference between high"</li><li>"Learn how to preprocess data" -&gt; "Learn how to preprocess text"</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/257/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (BenoitDherin) [inline notebooks/text_models/solutions/load_text.ipynb:1]: <p>Maybe add </p><p> </p><pre class="ql-syntax" spellcheck="false">pip freeze | grep tensorflow_text || pip install "tensorflow-text==2.8.*"
</pre><p> </p><p>or add the install in the top-level <code>requirements.txt</code> file </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/257/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (BenoitDherin) [inline notebooks/text_models/solutions/load_text.ipynb:1]: <p>Should we add this dataset to our public asl bucket instead?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/257/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (BenoitDherin) [inline notebooks/text_models/solutions/load_text.ipynb:1]: <p>Should we add this dataset to our public bucket?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/257/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='61'/>

**Review by BenoitDherin — APPROVED** [review]
LGTM. Great lab!

File: notebooks/text_models/solutions/load_text.ipynb (hunk 1 above)
Comment (sanjanalreddy) [reply notebooks/text_models/solutions/load_text.ipynb:1]: <p>Thank you! Updated to "Building NLP input pipelines with Keras and Tensorflow"</p>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/load_text.ipynb:1]: <p>Updated. Thank you! </p>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/load_text.ipynb:1]: <p>I'll add it to the requirements.txt to keep it similar to other labs.</p>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/load_text.ipynb:1]: <p>Updated the markdown. Thank you! </p>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/load_text.ipynb:1]: <p>Updated the instruction. <code>seed</code> is a required argument here for this function when specifying <code>validation_split</code> to ensure that there is no overlap between the training and validation subset . I updated the markdown and per Benoit's suggestion, also put it above.</p>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/load_text.ipynb:1]: <p>Updated the code:</p><ul><li>Converted the Question which was in bytes type to a string to make it more readable </li><li>Added a separator between each instance</li></ul>

[the remark this answers is not in the document]
Comment (sanjanalreddy) [reply notebooks/text_models/solutions/load_text.ipynb:1]: <p>We don't need to specify shuffle=False since we are using a seed. But if we don't use a seed then we have to specify <code>shuffle=False</code>.  <span class="ql-cursor">﻿</span></p>

[the remark this answers is not in the document]
Comment (sanjanalreddy) [reply notebooks/text_models/solutions/load_text.ipynb:1]: <p><code>subset</code> here doesn't take the value <code>both</code> unfortunately. It only takes <code>training</code> or <code>validation</code> . This is only used when we explicitly specify the validation split like we are doing above.</p>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/load_text.ipynb:1]: <p>Added this dataset to our public repo: <code>gs://asl-public/text/data</code></p>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/load_text.ipynb:1]: <p>Added this dataset to our public repo:&nbsp;<code>gs://asl-public/text/data</code></p>

[the remark this answers is not in the document]
Comment (sanjanalreddy) [reply notebooks/text_models/solutions/load_text.ipynb:1]: <p>Great catch! Thanks Takumi. I updated the code to reflect the new change.</p>

**Review by takumiohym — APPROVED** [review]
LGTM. Thanks for building a very practical notebook :)
